packages/react-aria-components/stories/Table.stories.tsx

```diff
@@ -393,6 +407,265 @@ const MyCheckbox = ({children, ...props}: CheckboxProps) => {
+
+export const RenderEmptyStateStory  = {
+  render: RenderEmptyState,
+  args: {
+    isLoading: false
+  },
+  name: 'Empty/Loading Table rendered with TableLoader collection element'
+};
+
+interface Character {
+  name: string,
+  height: number,
+  mass: number,
+  birth_year: number
+}
+
+const OnLoadMoreTable = () => {
+  let list = useAsyncList<Character>({
+    async load({signal, cursor}) {
+      if (cursor) {
+        cursor = cursor.replace(/^http:\/\//i, 'https://');
+      }
+
+      // Slow down load so progress circle can appear
+      await new Promise(resolve => setTimeout(resolve, 4000));
+      let res = await fetch(cursor || 'https://swapi.py4e.com/api/people/?search=', {signal});
+      let json = await res.json();
+      return {
+        items: json.results,
+        cursor: json.next
+      };
+    }
+  });
+
+  let isLoading = list.loadingState === 'loading' || list.loadingState === 'loadingMore';
+  let scrollRef = useRef<HTMLDivElement>(null);
+
+  return (
+    // TODO: but why typescript?
+    <ResizableTableContainer ref={scrollRef as RefObject<HTMLDivElement>} style={{height: 150, width: 400, overflow: 'auto'}}>
+      <Table aria-label="Load more table" isLoading={isLoading} onLoadMore={list.loadMore} scrollRef={scrollRef as RefObject<HTMLDivElement>}>
+        <TableHeader>
+          <Column id="name" isRowHeader style={{position: 'sticky', top: 0, backgroundColor: 'lightgray'}}>Name</Column>
+          <Column id="height" style={{position: 'sticky', top: 0, backgroundColor: 'lightgray'}}>Height</Column>
+          <Column id="mass" style={{position: 'sticky', top: 0, backgroundColor: 'lightgray'}}>Mass</Column>
+          <Column id="birth_year" style={{position: 'sticky', top: 0, backgroundColor: 'lightgray'}}>Birth Year</Column>
+        </TableHeader>
+        <MyTableBody
+          tableWidth={400}
+          renderEmptyState={() => renderEmptyLoader({isLoading: list.loadingState === 'loading', tableWidth: 400})}
+          isLoadingMore={list.loadingState === 'loadingMore'}
+          rows={list.items}>
+          {(item) => (
+            <Row id={item.name} style={{width: 'inherit', height: 'inherit'}}>
+              <Cell>{item.name}</Cell>
+              <Cell>{item.height}</Cell>
+              <Cell>{item.mass}</Cell>
+              <Cell>{item.birth_year}</Cell>
+            </Row>
+          )}
+        </MyTableBody>
+      </Table>
+    </ResizableTableContainer>
+  );
+};
+
+export const OnLoadMoreTableStory  = {
+  render: OnLoadMoreTable,
+  name: 'onLoadMore table'
+};
+
+// TODO: this doens't quite work because the table body isn't initially rendered thus the ref is scrollRef is undef
+// This could be remedied perhaps by moving the onLoadMore call to TableBody internally and passing down the scrollRef and other load props via context from body.
+// That would guarentee that the ref for the table body is defined at the time the hook renders. Stashed this change locally, to be discussed.
```

Comment:
i think earlier you said it'd match another API to move it down?

Reply:
The only that comes to mind is that moving loading to TableBody would match RSP more closely, but anyway for now the user will call this hook outside of table so this is inapplicable for now I guess (bug will still exist since the body ref isn't defined upon the first render but we can move this back in if it turns out we need the body to scroll I guess)